shell/platform/android/io/flutter/plugin/editing/SpellCheckPlugin.java

```diff
@@ -0,0 +1,135 @@
+    mTextServicesManager = textServicesManager;
+    mSpellCheckChannel = spellCheckChannel;
+
+    mSpellCheckChannel.setSpellCheckMethodHandler(this);
+  }
+
+  /**
+   * Unregisters this {@code SpellCheckPlugin} as the {@code
+   * SpellCheckChannel.SpellCheckMethodHandler}, for the {@link
+   * io.flutter.embedding.engine.systemchannels.SpellCheckChannel}, and closes the most recently
+   * opened {@code SpellCheckerSession}.
+   *
+   * <p>Do not invoke any methods on a {@code SpellCheckPlugin} after invoking this method.
+   */
+  public void destroy() {
+    mSpellCheckChannel.setSpellCheckMethodHandler(null);
+
+    if (mSpellCheckerSession != null) {
+      mSpellCheckerSession.close();
+    }
+  }
+
+  @Override
+  public void initiateSpellCheck(@NonNull String locale, @NonNull String text) {
+    performSpellCheck(locale, text);
+  }
+
+  /** Calls on the Android spell check API to spell check specified text. */
+  public void performSpellCheck(String locale, String text) {
+    String[] localeCodes = locale.split("-");
+    Locale localeFromString = LocalizationPlugin.localeFromString(locale);
+
+    if (mSpellCheckerSession != null) {
+      mSpellCheckerSession.close();
+    }
+    mSpellCheckerSession =
+        mTextServicesManager.newSpellCheckerSession(
+            null,
+            localeFromString,
+            this,
+            /** referToSpellCheckerLanguageSettings= */
+            true);
+
+    TextInfo[] textInfos = new TextInfo[] {new TextInfo(text)};
+    mSpellCheckerSession.getSentenceSuggestions(textInfos, MAX_SPELL_CHECK_SUGGESTIONS);
+  }
+
+  /**
+   * Callback for Android spell check API that decomposes results and send results through the
+   * {@link SpellCheckChannel}.
+   *
+   * <p>Spell check results will be encoded as a string representing the span of that result, with
+   * the format [start_index.end_index.suggestion_1,suggestion_2,suggestion_3], where there may be
+   * up to 5 suggestions.
+   */
+  @Override
+  public void onGetSentenceSuggestions(SentenceSuggestionsInfo[] results) {
+    ArrayList<String> spellCheckerSuggestionSpans = new ArrayList<String>();
+
+    if (results.length > 0) {
+      SentenceSuggestionsInfo spellCheckResults = results[0];
+
+      for (int i = 0; i < spellCheckResults.getSuggestionsCount(); i++) {
+
+        SuggestionsInfo suggestionsInfo = spellCheckResults.getSuggestionsInfoAt(i);
+        int suggestionsCount = suggestionsInfo.getSuggestionsCount();
+
+        if (suggestionsCount > 0) {
+          String spellCheckerSuggestionSpan = "";
+          int start = spellCheckResults.getOffsetAt(i);
+          int length = spellCheckResults.getLengthAt(i);
+
+          spellCheckerSuggestionSpan += (String.valueOf(start) + ".");
+          spellCheckerSuggestionSpan += (String.valueOf(start + (length - 1)) + ".");
+
+          for (int j = 0; j < suggestionsCount; j++) {
+            spellCheckerSuggestionSpan += (suggestionsInfo.getSuggestionAt(j) + ",");
```

Comment:
could the suggestion include any of the reserved characters? e.g. `,`, and `.`?  If they include those, then the code that parses the result might get very confused.

Reply:
I saw the test case,  using `.` to separate the offsets is ok. 

One possible solution would be to save the length of the word before we write the word. Since it's variable length encoding, we have no choice. Another possible solution, (simpler than encoding the length) is to use escape characters. e.g. new line. I don't think new line will be included in the suggestions.

Reply:
I believe punctuation would only be in results for grammar check, but I think it's worth fixing this now as I assume we will want to support grammar correction in the future (I plan to file a bug for that). So, at least for now, using new lines should be safe!